src/parse.coffee

```diff
@@ -32,6 +32,7 @@ parse = (source, callback) ->
 
     if apiElements
       apiElements.unshift(warning) if warning
+      apiElements.freeze() # Adds 'parent' properties, prevents mutation
```

Comment:
@kylef I'm thinking whether the `freeze()` method is intuitive or not. I sort of didn't feel all right using it without the explanatory comment. Freezing evokes immutability, that's fine, but it also adds parents, which isn't clear from the method call. Given it's called `freeze()` and it makes the thing immutable AND adds parents, it feels like adding parents is a magic "side effect". OTOH I couldn't come up with a better name.